### DISCUSSION TITLE
Raycasting scene and the underlying algorithm for signed distance queries

### DISCUSSION THREAD

**Discussion by JGittles** [isl-org/Open3D#5885] [Q&A] [answered]
Hi All,

While reading through the docs (http://www.open3d.org/docs/latest/tutorial/geometry/distance_queries.html#Distance-Queries), I was curious about how distance queries for meshes are calculated under the hood. My best guess is there is some sort of polyhedral geodesic distance function happening (such as https://www.cs.jhu.edu/~misha/ReadingSeminar/Papers/Mitchell87.pdf) but I was unsure why this needs this special scene construction. 

Really hoping to be able to see a paper for whatever method is used because I've been really impressed with the performance so far!

**Comment by benjaminum** [comment, marked as answer]
We use embree under the hood, which uses a bounding volume hierarchy for acceleration. The scene construction is necessary to create this data structure.

**Comment by JGittles** [comment]
Ah! I see this looks like a full-featured library. Thanks for the response! I'll mark this as answered. Also found what I assume you guys are interfacing with for point queries in their docs so I'll put that here for posterity: https://github.com/embree/embree#point-queries
